cmd/bootm.c

```diff
@@ -123,7 +123,7 @@ int do_bootm(cmd_tbl_t *cmdtp, int flag, int argc, char * const argv[])
 			return do_bootm_subcommand(cmdtp, flag, argc, argv);
 	}
 
-#ifdef CONFIG_IMX_HAB
+#if defined(CONFIG_IMX_HAB) && !defined(CONFIG_IMX8M)
```

Comment:
Can we have a separated IMX_HAB option that can be used here? This is not IMX8M specific. 

We can add a new option that is set by default if IMX_HAB and update this ifdef using the new option. Then we can easily disable the required option in our own config fragment.

Reply:
we could, but that could be done when support for another board comes along no? why should we try to be generic when there is only one board to support yet? board number 2 is the one that should define the interface IMO

Reply:
Because I don't see how this wouldn't be an issue with other targets, and best to fix in a generic way first.

Any iMX target that requires IMX_HAB to be enabled will face this same issue.

Reply:
well no, how would this be an issue for other targets? it hasnt been an issue so far so why now does it have to be an issue; of course it would be an issue if they try to use the functionality implemented for imx8mm since they dont have it yet.

I disagree with fixing in a generic way first. The first implementation should be specific and testable. the second implementation should be generic allowing the first implementer and the second implementer to tests separatelly.

Reply:
> well no, how would this be an issue for other targets? it hasnt been an issue so far so why now does it have to be an issue; of course it would be an issue if they try to use the functionality implemented for imx8mm since they dont have it yet.

What I mean is that it would not work for other targets as we enable a similar boot path, because of the same restrictions applied to imx8mm.

For example, CONFIG_IMX_HAB is required for mfgtools and this would cause an issue when loading the kernel fitimage as we do for imx7ulp (which is not an issue now as that target is using an older u-boot). Same will happen once we enable IMX_HAB on mfgtools for imx6ull (booting fit for installing at the linux level, and not u-boot). Same thing for imx7d.

Since we will have to fix this quite soon anyway, I would rather prefer to find a way to not imply HAB verified boot in bootm when IMX_HAB has to be enabled (otherwise we will have to come to the generic implementation in a few weeks).

> I disagree with fixing in a generic way first. The first implementation should be specific and testable. the second implementation should be generic allowing the first implementer and the second implementer to tests separatelly.

Sure, that works, but since we will have to fix for the other targets over the next few weeks, I would rather fix the generic way now and not have to come back to this later.

@mike-scott what do you think?